kernel/rtlil.h

```diff
@@ -657,38 +667,111 @@ namespace RTLIL
 
 struct RTLIL::Const
 {
-	int flags;
-	std::vector<RTLIL::State> bits;
+	short flags;
+	private:
+	typedef std::vector<RTLIL::State> bitvectype;
+	enum class backing_tag: bool { bits, string };
+	// Do not access the union or tag even in Const methods unless necessary
+	mutable backing_tag tag;
+	union {
+		mutable bitvectype bits_;
+		mutable std::string str_;
+	};
+
+	// Use these utilities instead
+	bool is_bits() const { return tag == backing_tag::bits; }
+	bool is_str() const { return tag == backing_tag::string; }
```

Comment:
Mark those private, possibly rename those, otherwise someone at one point will take those to be semantically significant (e.g. mistake them for a `flags & CONST_FLAG_STRING != 0` check)

I suggest `bits_backed()` and `str_backed()`

Reply:
They are private

Reply:
Ah the indentation threw me off, we should fix it